_posts/2025-01-03-Clang-17-now-in-dev-container.markdown

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+title:  "Clang 17 now in dev container, and other toolchain news"
+date:   2025-01-23
+categories: cheri toolchain
+author: Owen Anderson
+---
+
+Thanks to a collaboration with the [upstream CHERI toolchain](https://github.com/ctsrd-cheri/llvm-project), the CHERIoT toolchain has now been rebased onto Clang 17 from Clang 13, bringing us two years closer to upstream Clang.
+Major thanks to [Alex Richardson](https://github.com/arichardson) and [Sam Leffler](https://github.com/sleffler/) for their work on this effort!
+
+This updates brings with it substantially improved support for C++20 features, as well as preliminary support for some C++23 and C23 features.
+It also brings with it many improvements to the core of the RISC-V code generator, notably also benefiting code size: the firmware image for the Cheriot RTOS testsuite is 2.7% smaller when built with Clang 17 compared to Clang 13.
+Other highlights include compile time improvements, and too many under-the-hood fixes to enumerate.
+You can find detailed release notes for all Clang and LLVM releases on the LLVM.org [releases page](https://releases.llvm.org).
+
```

Comment:
Might be worth reminding anyone using the dev container that they'll need to pull the latest version.  Presumably builds will fail if its not aligned ?

Reply:
We've tried to make sure that the RTOS works with the old compiler, but explicitly updating is recommended.

Reply:
Done

Reply:
The only place that mattered was Sail.  We've merged the change to sail, so if you update the dev container to get the new compiler, you'll also get the new sail.  Older sail needs the -m flag to be passed.  Might be good to mention if people are building things themselves.

Reply:
Note added